### PR TITLE
Add `InstantLoan`

### PR DESCRIPTION
**Pull request by kasperbirch1**
#### Link to issue
https://reload.atlassian.net/browse/DSC-11

#### This PR is submitted to introduce the "InstantLoan" feature

The `InstantLoan` feature required changes to the `Disclosure` and `AvailabilityLabel` components. Specifically, the Disclosure component now accepts a new prop called 'summary' to support the unique summary of the InstantLoan feature. Additionally, a new component called `AvailabilityLabelVisual` was added to support the InstantLoan feature.

In addition, I have added the `InstantLoan` component and the associated logic for how and when the component should be rendered

These additions allow users to view and access instant loan branches in `ReservationModalBody` if the material has available instant loans 

#### Screenshot of the result

image

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

Please note that this pull request depends on https://github.com/danskernesdigitalebibliotek/dpl-react/pull/385, and it should be merged after that pull request has been merged.

### CODE REVIEW

**Review by spaceo — COMMENTED** [review]
Apart from the refactoring we have talked about I have some other remarks

File: src/components/Disclosures/disclosure.dev.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { ComponentStory, ComponentMeta } from "@storybook/react";
```
Comment (spaceo) [inline src/components/Disclosures/disclosure.dev.tsx:1]: Just double checkking:
Do you deliberately want to commit this story?
The story is copying what the MaterialDisclosure is doing

File: src/components/availability-label/availability-label-inside.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          !selected
+      }
+    ),
+    check: clsx("availability-label__check", selected && "selected")
+  };
+
+  return (
+    <>
+      <div className={classes.triangle} />
+      <img className={classes.check} src={CheckIcon} alt="check-icon" />
```
[the remark this answers is not in the document]
Comment (sinejespersen) [reply src/components/availability-label/availability-label-inside.tsx:36]: Assuming this is decorative?

File: src/apps/material/MaterialDisclosure.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  icon: string;
+  children?: ReactNode;
+}
+
+const MaterialDisclosure: FC<MaterialDisclosureProps> = ({
+  title,
+  icon,
+  children,
+  dataCy
+}) => {
```
[the remark this answers is not in the document]
Comment (kasperbirch1) [reply src/apps/material/MaterialDisclosure.tsx:17]: Yes, you're correct!  This pattern should make it more readable and reusable 😊

**Review by sinejespersen — APPROVED** [review]

**Pull request merged**